### PR TITLE
Fix Bluetooth share without breaking others

### LINKED ISSUE #1223: Key shared via bluetooth is encoded in HTML
When I tried to copy my key to my Mac via Bluetooth, the Mac received a file named "bluetooth_content_share.html" with contents that looked like this:

    <html>
    <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    </head>
    <body>
    -----BEGIN PGP PUBLIC KEY BLOCK-----<br>
    <br>
    gVMEc6PwyOL63szYP1idG6G2k93azdbrczHK5xDNAxCjYRdZDfYsTEO7s<a href="tel:118">118</a>uEhA<br>
    jzBKVNFQD/B00js83AG97sOHsboc39liyxG8PE4jgiuHIYYSmy2BAJoG8YDxvpuy<br>
    F6E5dBlUCYMDQXF9TqluF1aXDpenYKVr6W4Rckvu1fhB6CM4LIjBuDdQju5EIfJT<br>
    ⋮
    XgwNH97I58vNgTtwueVnM96w2AXGMrfQRMPluGm22GBeF1LxcAJAUiHjvE8VQomZ<br>
    jI/BG0y5OQqsq+GzTNkqtg<br>
    -----END PGP PUBLIC KEY BLOCK-----<br>
    </body>
    </html>

The data presented here is bogus, but the html structure is the same, except that the actual file had no newlines in it, not even a trailing one.  Note that any sequence of 3 or more decimal digits got converted to a "tel" url-scheme link.

I doubt that OpenKeychain is creating this HTML version; I'm guessing it probably needs to tell the Bluetooth sharing facility in Android not to mangle the file like this.

### PR DESCRIPTION
**Pull request by wfaulk**
Implemented Bluetooth sharing via file while leaving all other sharing alone.

Fixes #1223

### CODE REVIEW

**Comment by dschuermann** [comment]
Nice work!
Instead of using another caching implementation, please use TemporaryStorageProvider (BTW, which needs to be fixed, see https://github.com/open-keychain/open-keychain/issues/1241).
Also, I would advise that we use the primary user id as a filename instead of the short key id. This is much more descriptive, short key ids serve no real purpose and should not be used anymore, see https://github.com/open-keychain/open-keychain/wiki/OpenPGP-Security#key-ids-arent-displayed .

@wfaulk Do you have time to do these changes? Would be really great!

**Comment by wfaulk** [comment]
Had to rebase to get this to merge.  It should be good, though.

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/ViewKeyAdvShareFragment.java [hunk 1]
```diff
@@ -213,13 +219,48 @@ private void share(Uri dataUri, ProviderHelper providerHelper, boolean fingerpri
...
+                // Bluetooth Share will convert text/plain sent via EXTRA_TEXT to HTML
+                // Add replacement extra to send a text/plain file instead.
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                    try {
+                        String primaryUserId = UncachedKeyRing.decodeFromData(content.getBytes()).
+                                getPublicKey().getPrimaryUserIdWithFallback();
+
+                        TemporaryStorageProvider shareFileProv = new TemporaryStorageProvider();
+                        Uri contentUri = TemporaryStorageProvider.createFile(getActivity(),
+                                primaryUserId + Constants.FILE_EXTENSION_PGP_ALTERNATE + Constants.FILE_EXTENSION_ASC);
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/ViewKeyAdvShareFragment.java]: only use one file ending, in this case Constants.FILE_EXTENSION_ASC

**Comment by dschuermann** [comment]
The changes in KeychainApplication and KeyFormattingUtils shouldn't be required any more, please revert them.

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/ViewKeyAdvShareFragment.java (hunk 1 above)
Comment (wfaulk) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/ViewKeyAdvShareFragment.java]: Done.

**Comment by wfaulk** [comment]
Whoops.  Fixed.

**Comment by dschuermann** [comment]
Great thanks!

**Pull request merged**